addons/account/models/chart_template.py

```diff
@@ -217,11 +217,18 @@ def _pre_reload_data(self, company, template_data, data):
         for xmlid, journal_data in list(data.get('account.journal', {}).items()):
             if self.ref(xmlid, raise_if_not_found=False):
                 del data['account.journal'][xmlid]
-            elif 'code' in journal_data:
-                journal = self.env['account.journal'].with_context(active_test=False).search([
-                    ('code', '=', journal_data['code']),
-                    ('company_id', '=', company.id),
-                ])
+            else:
+                if 'code' in journal_data:
+                    journal = self.env['account.journal'].with_context(active_test=False).search([
+                        ('code', '=', journal_data['code']),
+                        ('company_id', '=', company.id),
+                    ])
+                if not journal and 'name' in journal_data and 'type' in journal_data:
```

Comment:
Add a comment mentioning it is to cover the unique constraint on the alias

Reply:
comment added